File: backend/src/domain/entities/LearningResource.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
} from 'typeorm';

@Entity('learning_resources')
export class LearningResource {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'organization_id' })
  organizationId: number;

  @Column({ name: 'uploader_member_id' })
  uploaderMemberId: number;

  @Column({ type: 'varchar', nullable: true })
  title: string;

  @Column({ type: 'varchar', nullable: true })
  type: string;

  @Column({ type: 'varchar', name: 'link_url', nullable: true })
  linkUrl: string;

  @Column({ type: 'text', nullable: true })
  description: string;

  @CreateDateColumn({ type: 'timestamp', name: 'created_at' })
  createdAt: Date;
}

File: backend_new_structure/common/src/main/java/com/service/common/entity/LearningResource.java
package com.service.common.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("learning_resources")
public class LearningResource {

    @Id
    private Long id;

    @Column("organization_id")
    private Long organizationId;

    @Column("uploader_member_id")
    private Long uploaderMemberId;

    private String title;

    private String type;

    @Column("link_url")
    private String linkUrl;

    private String description;

    @CreatedDate
    @Column("created_at")
    private LocalDateTime createdAt;
}
